test/parallel/rolldown.withdraw.merkleRoot.test.ts

```diff
@@ -0,0 +1,472 @@
+      Assets.mintToken(ETH_ASSET_ID, testUser, BN_MILLION),
+    );
+    await signTx(
+      getApi(),
+      await Withdraw(testUser, BN_HUNDRED, ethIdL1Asset.ethereum, "Ethereum"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    const event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const l2Request = await Rolldown.getL2Request(event.range.to.toNumber());
+    expect(l2Request.withdrawal.tokenAddress).toEqual(ethIdL1Asset.ethereum);
+    expect(l2Request.withdrawal.withdrawalRecipient).toEqual(
+      testUser.keyRingPair.address,
+    );
+  });
+
+  test("Given a withdraw in a batch extrinsic, When one of the calls fail from the batch, then tokens are reverted ( transactional )", async () => {
+    const nonExistedToken = await getNonExistedTokenId();
+    const events = await Sudo.batchAsSudoFinalized(
+      ...(await getWithdrawalExtrinsics(10)),
+      await Withdraw(
+        testUser,
+        10,
+        nonExistedToken.toHuman().ethereum,
+        "Ethereum",
+      ),
+    );
+    const error = events.filter(
+      (x) => x.method === "ExtrinsicFailed" && x.section === "system",
+    );
+    expect(error.length).toBe(1);
+    expect(error[0].error!.name).toBe("TokenDoesNotExist");
+    await testUser.refreshAmounts(AssetWallet.AFTER);
+    const withdrawalAmount = testUser
+      .getAsset(ETH_ASSET_ID)!
+      .amountBefore.free.sub(testUser.getAsset(ETH_ASSET_ID)!.amountAfter.free);
+    expect(withdrawalAmount).bnEqual(BN_ZERO);
+  });
+
+  test("Given <3> withdrawals WHEN they run successfully and we wait <batchPeriod> blocks, a batch is created upon reaching the period", async () => {
+    await Sudo.batchAsSudoFinalized(...(await getWithdrawalExtrinsics(3)));
+    const event = await Rolldown.waitForNextBatchCreated(
+      "Ethereum",
+      waitingBatchPeriod,
+    );
+    const sequencersList = await SequencerStaking.activeSequencers();
+    expect(sequencersList.toHuman().Ethereum).toContain(event.assignee);
+    expect(event.source).toEqual("PeriodReached");
+  });
+
+  test("Given <1> withdrawal WHEN we manually generate a batch for it THEN a batch is generated", async () => {
+    await signTx(
+      getApi(),
+      await Withdraw(testUser, BN_HUNDRED, gaspIdL1Asset.ethereum, "Ethereum"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    await signTx(
+      getApi(),
+      await Rolldown.createManualBatch("EthAnvil"),
+      testUser.keyRingPair,
+    ).then((events) => {
+      const res = getEventResultFromMangataTx(events);
+      expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+    const event = await Rolldown.waitForNextBatchCreated(
```

Comment:
this is wrong, batch must be already been generated on "Manual batch" Tx, plz review. 
Events must come from the events there, no need for waiting.

Reply:
We need to receive event, so I changed waiting period from batchPeriod to 0. Done